### PR TITLE
adds fun!

### PR DESCRIPTION
**Pull request by Kmc2000**
## About The Pull Request

nothing

## Why It's Good For The Game

microwave/proc/wzhzhzhzhzhz()

## Changelog
:cl:
add: Added new things
/:cl:

### CODE REVIEW

File: nsv13/code/game/gamemodes/pvp/base_defense.dm [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+//Syndie PVP ship
+
+/area/tartarus
+	name = "Tartarus"
+	ambientsounds = list('nsv13/sound/ambience/leit_motif.ogg','nsv13/sound/ambience/wind.ogg','nsv13/sound/ambience/wind2.ogg','nsv13/sound/ambience/wind3.ogg','nsv13/sound/ambience/wind4.ogg','nsv13/sound/ambience/wind5.ogg','nsv13/sound/ambience/wind6.ogg')
+	noteleport = TRUE
+	icon_state = "syndie-ship"
+	has_gravity = TRUE
+
+/area/tartarus/underground
```
Comment (francinum) [inline nsv13/code/game/gamemodes/pvp/base_defense.dm:113]: Refactoring this to a sub-type of the mining area and causing it to handle cave generation would likely help with the lack of underground flavor.

File: nsv13/code/game/gamemodes/pvp/base_defense.dm [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+	ai_controlled = FALSE
+	role = PVP_SHIP
+
+//Overridable method which spawns in the ship we want to use.
+/datum/game_mode/pvp/base_defense/generate_ship(ship_type, map_file)
+	ship_type = /obj/structure/overmap/planet/pvp //Yes. I'm a hack. Sue me Fran.
+	return instance_overmap(_path=ship_type, folder= "map_files/PVP" ,interior_map_files = map_file, traits=list(list(ZTRAIT_UP=1, ZTRAIT_LINKAGE = SELFLOOPING, ZTRAIT_BOARDABLE = 1),list(ZTRAIT_DOWN = -1, ZTRAIT_LINKAGE = SELFLOOPING, ZTRAIT_BOARDABLE = 1)))
+
+
+/datum/game_mode/pvp/base_defense/force_lighting(obj/structure/overmap/hammurabi)
```
Comment (francinum) [inline nsv13/code/game/gamemodes/pvp/base_defense.dm:201]: Fucking hell. There has to be *some* way to do this that doesn't involve a waitfor-false proc and *multiple sleeps*

File: nsv13/code/game/gamemodes/pvp/base_defense.dm [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+	var/baseturf = /turf/open/floor/plating/asteroid/basalt
+	var/list/baseturf_to_replace = typecacheof(/turf/open/space)
+	baseturf_to_replace += typecacheof(/turf/open/openspace)
+	for(var/area/AR in hammurabi.linked_areas) //Fucking force a lighting update IDEK why we have to do this but it just works
+		if(AR.dynamic_lighting)
+			AR.set_dynamic_lighting(DYNAMIC_LIGHTING_DISABLED)
+			sleep(1)
+			AR.set_dynamic_lighting(DYNAMIC_LIGHTING_ENABLED)
+		//Baseturf editors didn't do the trick. Time to take matters into my own hands...
+		for(var/turf/thing in get_area_turfs(AR))
```
Comment (francinum) [inline nsv13/code/game/gamemodes/pvp/base_defense.dm:212]: *V O M I T*

**Comment by Kmc2000** [comment]
> `_maps/map_files/PVP/ice_colony.dmm` must be split-Z'd.
> I acknowledge that this may complicate loading.

Oh yeah forgot about that

Nah it's all good, instance overmap can handle multiple map files

**Comment by francinum** [comment]
@Kmc2000 does this still need to be open?

**Comment by francinum** [comment]
Rest in peace sweet prince.

**Comment by QAI-07** [comment]
FREEMAN, YOU FOOL!

**Comment by Bokkiewokkie** [comment]
Planetary assault 2 electric boogaloo??

**Comment by Pontenerd** [comment]
Waltuh, reclose the pr waltuh

**Comment by sunofang** [comment]
OH GOD OH FUCK OH GOD

**Comment by Kenionatus** [comment]
F

**Pull request closed without merging**